### PR TITLE
Avoid using `get_plugins_manager` method

### PR DESCRIPTION
**Pull request by IKarbowiak**
Provide plugin manager as a parameter where it's possible.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [x] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [x] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [x] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-1976 Optimize getting plugin menager](https://app.clickup.com/t/2549495/SALEOR-1976)

**Review by korycins — APPROVED** [review]

**Review by d-wysocki — APPROVED** [review]

File: saleor/payment/gateway.py [hunk 1]
```diff
@@ -76,12 +74,10 @@ def wrapped(payment: Payment, *args, **kwargs):
 def process_payment(
     payment: Payment,
     token: str,
+    plugin_manager: "PluginsManager",
```
Comment (maarcingebala) [inline saleor/payment/gateway.py:77]: I think in most places in the codebase this argument is called `manager`. I would rename it to `manager` here and in other functions that use a different name.

File: saleor/product/tests/test_product_availability.py [hunk 2]
```diff
@@ -22,12 +22,14 @@ def test_availability(stock, monkeypatch, settings, channel_USD):
...
         PluginsManager, "apply_taxes_to_product", Mock(return_value=taxed_price)
     )
+    manager = get_plugins_manager()
     availability = get_product_availability(
         product=product,
         product_channel_listing=product_channel_listing,
         variants=product.variants.all(),
         variants_channel_listing=variants_channel_listing,
         channel=channel_USD,
+        plugins=manager,
```
Comment (maarcingebala) [inline saleor/product/tests/test_product_availability.py:32]: In `get_product_availability` we could also rename `plugins` to `manager` for consistency.

**Review by maarcingebala — COMMENTED** [review]
Added two comments about making naming more consistent. I think in the future we should also rename `info.context.plugins` to `info.context.manager` for consistency, but we need to discuss it first and it should be done in separate PR.

**Review by fowczarek — APPROVED** [review]

**Review by maarcingebala — APPROVED** [review]

**Pull request merged**